projects/plugins/jetpack/extensions/shared/memberships/utils.js

```diff
@@ -23,7 +24,14 @@ export const encodeValueForShortcodeAttribute = value => {
 };
 
 export const getPaidPlanLink = alreadyHasTierPlans => {
-	const link = 'https://wordpress.com/earn/payments/' + location.hostname;
+	const isJetpackSite = ! isAtomicSite() && ! isSimpleSite();
+
+	const base = isJetpackSite
+		? 'https://cloud.jetpack.com/monetize/payments/'
+		: 'https://wordpress.com/earn/payments/';
+
+	const link = base + location.hostname;
```

Comment:
This may be a good opportunity to move away from `location.hostname`, which won't return the correct information on sites that live in a subdirectory.

Could we rely on `getRedirectUrl()` here maybe?

Reply:
Looks like a combo of `getRedirectUrl()` and `getSiteFragment()` would be best, and there are other places with /earn links where this is needed. Like Donations block.